### PR TITLE
fix: replace "owner" label with "space" label

### PR DESCRIPTION
**Pull request by mfrancisc**
## Description
This PR adds `SpaceLabelKey` that will be set on the Namespace and namespace/cluster-scoped resources.
It will replace the `OwnerLabelKey` usage.

Jira: https://issues.redhat.com/browse/CRT-1776

## Checks
1. Did you run `make generate` target? **yes**

2. Did `make generate` change anything in other projects (host-operator, member-operator)? **no**

3. In case of **new** CRD, did you the following? **N/A**

4. In case other projects are changed, please provides PR links.
    - member-operator: https://github.com/codeready-toolchain/member-operator/pull/438

### CODE REVIEW

File: api/v1alpha1/nstemplateset_types.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 
 const (
 	OwnerLabelKey       = LabelKeyPrefix + "owner"
+	SpaceLabelKey       = SpaceBindingSpaceLabelKey
```
Comment (mfrancisc) [inline api/v1alpha1/nstemplateset_types.go:9]: or maybe we could switch:
```
SpaceBindingSpaceLabelKey = SpaceLabelKey
``
?
🤔

Comment (mfrancisc) [reply api/v1alpha1/nstemplateset_types.go:9]: Yeah, I thought about that at first, but I didn't want to go over a bigger refactor that includes also renaming `SpaceBindingSpaceLabelKey` occurrences with `SpaceLabelKey` .  But I can do that of course.

Comment (rajivnathan) [reply api/v1alpha1/nstemplateset_types.go:9]: Makes sense, if it's too much refactoring then `SpaceBindingSpaceLabelKey = SpaceLabelKey` works too 👍

Comment (mfrancisc) [reply api/v1alpha1/nstemplateset_types.go:9]: The refactor by itself it's easy to do (with the fancy refactor IDE functionality 😄  ) , but I would have to open a PR for this in: regservice, host-operator, toolchain-common, toolchain-e2e and sandbox-sre. 
Maybe I can do this as an afterwards cleanup. WDYT ?

Comment (rajivnathan) [reply api/v1alpha1/nstemplateset_types.go:9]: I didn't look at all the places `SpaceBindingSpaceLabelKey` is used and opening multiple PRs is time consuming so if it requires opening PRs in all of those repos as a separate task then it's probably more work than is necessary for not much benefit so it's probably better to do as you suggested instead.

**Review by rajivnathan — APPROVED** [review]

File: api/v1alpha1/nstemplateset_types.go (hunk 1 above)
Comment (alexeykazakov) [reply api/v1alpha1/nstemplateset_types.go:9]: Yeah.. I would stick with `SpaceBindingSpaceLabelKey = SpaceLabelKey` for now and maybe get rid of `SpaceBindingSpaceLabelKey` later.

**Review by alexeykazakov — APPROVED** [review]

File: api/v1alpha1/nstemplateset_types.go (hunk 1 above)
Comment (mfrancisc) [reply api/v1alpha1/nstemplateset_types.go]: Changed in  https://github.com/codeready-toolchain/api/pull/359/commits/ebf567185f004fa6be2bbfd4bc501c0168bac26a

**Review by MatousJobanek — APPROVED** [review]

**Review by xcoulon — APPROVED** [review]

**Pull request merged**